index.js

```diff
@@ -31,6 +41,31 @@ var createPluginFactory = function(options) {
         });
     }
 
+    return {
+        registratorName: registratorName,
+        runtime: runtime,
+        blackbox: blackbox,
+        basePath: basePath
+    };
+};
+
+var createPluginFactory = function(options) {
+    var areOptionsSet = false;
+    var registratorName;
+    var runtime;
+    var basePath;
+    var blackbox;
+
+    if (options) {
+      options = normalizeOptions(options);
```

Comment:
I think this fragment could be transformed to function 

```js
function applyOptions(options) {
  if (!options) {
    return;
  }

  options = normalizeOptions(options);
  registratorName = options.registratorName;
  runtime = options.runtime;
  basePath = options.basePath;
  blackbox = options.blackbox;

  applyOptions = function() {};
}

applyOptions(options);
```

and inside Program:

```js
applyOptions(getPluginOpts(file));
```